frontend/src/components/TypeaheadSelect.tsx

```diff
@@ -440,26 +520,21 @@ const TypeaheadSelect: React.FunctionComponent<TypeaheadSelectProps> = ({
         {...props}
       >
         <SelectList>
-          {filteredSelections.map((option, index) => {
-            const { content, value, dropdownLabel, ...optionProps } = option;
+          {(() => {
+            let idx = 0;
+            const groupOpts = Object.entries(groupedSelections.group).map(([groupName, group]) => {
+              const { node, nextIndex } = tGroupOption(groupName, group, idx);
+              idx = nextIndex;
+              return node;
+            });
+            const selectOpts = groupedSelections.noGroup.map((opt) => tSelectOption(opt, idx++));
             return (
-              <SelectOption
-                key={value}
-                value={value}
-                isFocused={focusedItemIndex === index}
-                {...optionProps}
-              >
-                {dropdownLabel ? (
-                  <Flex>
-                    <FlexItem>{content}</FlexItem>
-                    <FlexItem>{dropdownLabel}</FlexItem>
-                  </Flex>
-                ) : (
-                  content
-                )}
-              </SelectOption>
+              <>
+                {groupOpts}
+                {selectOpts}
+              </>
             );
-          })}
+          })()}
```

Comment:
nit: invoking function inline like this. Id rather have it moved out into a function above the render